pkg/flag/remote_flags.go

```diff
@@ -87,14 +87,14 @@ func (f *RemoteFlags) ToOptions() RemoteOptions {
 	token := viper.GetString(ServerTokenFlag)
 	tokenHeader := viper.GetString(ServerTokenHeaderFlag)
 
-	if serverAddr == "" {
+	if serverAddr == "" && listen == "" {
 		switch {
 		case len(lo.FromPtr(f.CustomHeaders)) > 0:
-			log.Logger.Warn(`"--custom-header"" can be used only with "--server"`)
-		case token != "" && listen == "":
-			log.Logger.Warn(`"--token" can be used only with "--server"`)
+			log.Logger.Warn(`"--custom-header" can be used only with "--server" or with "server" subcommand`)
+		case token != "":
+			log.Logger.Warn(`"--token" can be used only with "--server" or with "server" subcommand`)
```

Comment:
The `server` subcommand always set `--listen`, so it will never reach here as you added `listen == ""` at line 90.
```suggestion
			log.Logger.Warn(`"--custom-header" can be used only with "--server" or with "server" subcommand`)
		case token != "":
			log.Logger.Warn(`"--token" can be used only with "--server"`)
```